code/game/objects/structures/crates_lockers/closets/cardboardbox.dm

```diff
@@ -38,25 +42,35 @@
 	move_delay = FALSE
 
 /obj/structure/closet/cardboard/open(mob/living/user, force = FALSE)
-	if(opened || !can_open(user, force))
-		return FALSE
-	var/list/alerted = null
-	if(egged < world.time)
-		var/mob/living/Snake = null
-		for(var/mob/living/L in src.contents)
-			Snake = L
-			break
-		if(Snake)
-			alerted = viewers(7,src)
-	..()
-	if(LAZYLEN(alerted))
-		egged = world.time + SNAKE_SPAM_TICKS
-		for(var/mob/living/L in alerted)
-			if(!L.stat)
-				if(!L.incapacitated(IGNORE_RESTRAINTS))
-					L.face_atom(src)
-				L.do_alert_animation()
-		playsound(loc, 'sound/machines/chime.ogg', 50, FALSE, -5)
+	var/do_alert = (COOLDOWN_FINISHED(src, alert_cooldown) && (locate(/mob/living) in contents))
+
+	if(!do_alert)
+		return ..()
+
+	// Cache the list before we open the box.
+	var/list/alerted = viewers(7, src)
+
+	// There are no mobs to alert?
+	if(!(locate(/mob/living) in alerted))
+		return ..()
+
+	. = ..()
+
+	// Box didn't open?
+	if(!.)
+		return
```

Comment:
is this really the best way to go about this? Can't we check to open the box first by calling parent, then further down we can just early return without ..()? Also, do we need the locate early return when do_alert already checks that?

Reply:
Locate in contents is different from locate in alerted.

If there was nobody in the contents, we don't want to alert and can early return parent as its just a normal box at that point. This stops a viewers call being necessary just below.

We then see if there are any living mob viewers to alert. If there are not, another early return.

The reason the parent call is delayed is because it mutates srcs state. When you open the box via parent call first, how do you know there was a mob inside the box before it was opened?